src/pages/StableSwap/index.tsx

```diff
@@ -462,59 +486,59 @@ export default function StableSwap() {
-                        //     swapErrorMessage: undefined,
-                        //     showConfirm: true,
-                        //     txHash: undefined
-                        //   })
-                        // }
+                        if (isExpertMode) {
+                          handleSwap()
+                        } else {
+                          setSwapState({
+                            tradeToConfirm: trade,
+                            attemptingTxn: false,
+                            swapErrorMessage: undefined,
+                            showConfirm: true,
+                            txHash: undefined
+                          })
+                        }
                       }}
                       width="48%"
                       id="swap-button"
                       disabled={
-                        !isValid || approval !== ApprovalState.APPROVED || (priceImpactSeverity > 3 && !isExpertMode)
+                        !isValid ||
+                        approval !== ApprovalState.APPROVED ||
+                        (priceImpactSeverity === true && !isExpertMode)
                       }
-                      error={isValid && priceImpactSeverity > 2}
+                      error={isValid && priceImpactSeverity === true}
                     >
                       <Text fontSize={16} fontWeight={500}>
-                        {priceImpactSeverity > 3 && !isExpertMode
+                        {priceImpactSeverity === true && !isExpertMode
                           ? t('swapPage.priceImpactHigh')
-                          : t('swapPage.swap') + `${priceImpactSeverity > 2 ? t('swapPage.anyway') : ''}`}
+                          : t('swapPage.swap') + `${priceImpactSeverity === true ? t('swapPage.anyway') : ''}`}
                       </Text>
                     </ButtonError>
                   </RowBetween>
                 ) : (
                   <ButtonError
                     onClick={() => {
-                      // @nocommit -- this removes the confirmation modal
-                      // if (isExpertMode) {
-                      handleSwap()
-                      // } else {
-                      //   setSwapState({
-                      //     tradeToConfirm: trade,
-                      //     attemptingTxn: false,
-                      //     swapErrorMessage: undefined,
-                      //     showConfirm: true,
-                      //     txHash: undefined
-                      //   })
-                      // }
+                      if (isExpertMode) {
+                        handleSwap()
+                      } else {
+                        setSwapState({
+                          tradeToConfirm: trade,
+                          attemptingTxn: false,
+                          swapErrorMessage: undefined,
+                          showConfirm: true,
+                          txHash: undefined
+                        })
+                      }
```

Comment:
Could this become a callback fn since it's the same as 489-499?

Reply:
as in refactor to extract out and dry?